assets/js/core/WakaTimeCore.js

```diff
@@ -137,6 +134,50 @@ class WakaTimeCore {
     }
 
     /**
+     * Creates an array from list using \n as delimiter
+     * and checks if any element in list is contained in the url.
+     * Also checks if element is assigned to a project using @@ as delimiter
+     *
+     * @param url
+     * @param list
+     * @returns {object}
+     */
+    getHeartbeat(url, list) {
+        var lines = list.split('\n');
+
+        for (var i = 0; i < lines.length; i ++) {
+            // Trim all lines from the list one by one
+            var cleanLine = lines[i].trim();
+
+            // If by any chance one line in the list is empty, ignore it
+            if (cleanLine === '') {
+                continue;
+            }
+
+            // If url contains the current line return object
+            if (url.indexOf(cleanLine.split('@@')[0]) > -1) {
+                if (cleanLine.split('@@')[1]) {
+                    return {
+                        url: cleanLine.split('@@')[0],
+                        project: cleanLine.split('@@')[1]
+                    };
+                }
+                else {
+                    return {
+                        url: cleanLine.split('@@')[0],
+                        project: false
```

Comment:
I'll change this after merging, but we should use `null` instead of `false` because `null` evaluates to false but if sent to the API a `null` will be accepted as an empty value while `false` will be rejected as an invalid type for a string.

Reply:
I thought about that. When project is false (or null), then the project is replaced by '\<\<LAST_PROJECT\>\>' string in _preparePayload method. API always receives a string from the extension.

```js
_preparePayload(heartbeat, type, debug = false) {
    return JSON.stringify({
        entity: heartbeat.url,
        type: type,
        time: moment().format('X'),
        project: heartbeat.project || '<<LAST_PROJECT>>',
        is_debugging: debug,
        plugin: 'chrome-wakatime/' + config.version
    });
}
```